news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,471 @@
+
+### No clear vision
+
+To me, it looks like PrestaShop 1.7.0 was developed and released without a clear, global vision of what the ultimate objective of this major version was, and how long it would take to do it. 
+
+In addition, a tradition of word-of-mouth knowledge sharing at PrestaShop meant that if there ever was a vision, it was unknown to most and is now forever lost. This is something we are meaning to change, and this series is part of that.
+
+### Too many new things
+
+Too many technical projects were started in parallel, and were released before being properly finished.
+ 
+As I said before, this would have been acceptable in a SaaS software where you can work iteratively and release often, but doesn't work well for a development platform like PrestaShop, where upgrades are costly and infrequent. Unsurprisingly, releasing this way resulted in a giant, arguably well-deserved backlash from the community, who described PrestaShop 1.7 as "unfinished software".
+
+Here's an example. Back in 2017, we announced that we had started [introducing Vue JS in the Back Office][vuejs], supported by a BO API, which would be used in the Stocks and Translations pages.
+
+While I think this is the way of the future (no spoilers!), in hindsight, I believe that it was too early to introduce it in 1.7, and will probably have to be redone.
+
+Don't start screaming at me just yet! Let me explain why.
+
+First of all, if you take a glance at the [architecture overview][comprehensive-overview], you'll notice that there are two different stacks in the Back Office: the _Legacy stack_ and the _New stack_. The former is pretty much unchanged since 1.6, while the latter is essentially made out of Symfony controllers and Symfony forms, rendered with twig. Two stacks, one that will progressively replace the other. Right? 
+
+**Wrong.** If you look closer, you'll see that the _New stack_ actually has not one, but **two** implementations: the Symfony one that I described, and another one, based on VueJS and an API.
+
+So, we don't have _two_ different ways of doing things, we have _three_. 
+
+This is because the initial idea was that VueJS frontend and API backend was supposed to be the way the migration to Symfony should have continued starting on 1.7.3.0. But we chose to put that on hold and continue working with Symfony forms and Twig. Why?
+
+Because of _customization_.
+
+PrestaShop owes its success mainly to its ability to be extended and customized. For this, developers rely on extension points that allow them to add behavior in specific points, as well as conventions that allow them to replace certain bits of the software by their own.
```

Comment:
We can quote https://devdocs.prestashop.com/1.7/modules/concepts/ here. Not sure if it's a good idea though.

Reply:
I think that article should be improved because it only talks about Symfony features.

Reply:
Indeed, and that's a mistake. I fixed it in https://github.com/PrestaShop/docs/pull/353 we should see it in production soon

Reply:
It's deployed 😊https://devdocs.prestashop.com/1.7/modules/concepts/

Reply:
Done